### DISCUSSION TITLE
docs/cli/gen-pbentity

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#77] [General]
# docs/cli/gen-pbentity

使用GoFrame框架的命令行工具gf来根据数据库表生成proto数据结构文件pbentity。包括命令使用方法、选项配置及其说明，以及与gen dao模块中生成的entity文件的区别。适用于HTTP和GRPC服务的数据实体结构生成，支持多种数据库的生成规则配置。

https://goframe.org/docs/cli/gen-pbentity

**Comment by FireTiger23** [comment]
这个是否也能加一个 tablesEx 参数 来排除某些表？

**Reply by gqcn** [reply]
这个可以有，感兴趣可以提交PR参与共建。

**Comment by korialuo** [comment]
2.8.1版本 gf 工具。 hack/config.yaml 里配置了 gfcli.gen.pbentity.option: option go_package = "xxxxx/api/pbentity" (默认是 "/api/pbentity" 导致编译不通过) 后，并没覆盖默认的 option go_package.

**Reply by gqcn** [reply]
是的，没法覆盖已有的，只能`extra`额外的`option`

**Reply by 2265552611** [reply]
pbentity没有自动识别go.mod的包名，hack的配置也不生效，怎么生成都是/api/pbentity

**Reply by korialuo** [reply]
目前我是通过在hack文件里手动配置go_package解决的。

**Comment by llorz-o** [comment]
如何在生成时排除某些表里的字段，例如密码字段，这样我在其他 proto 中使用结构时无需返回该字段
